### PR TITLE
RSDK-474 Save resource error when not started

### PR DESCRIPTION
**Pull request by cheukt**
saves the error to the wrapper, which is returned when we try to access that resource.

Also updated some errors and logging to always return resource name + model if applicable

### CODE REVIEW

File: robot/impl/local_robot.go [hunk 1]
```diff
@@ -554,7 +554,7 @@ func (r *localRobot) newResource(ctx context.Context, config config.Component) (
 	}
 
 	if err != nil {
-		return nil, errors.Errorf("error building resource %s/%s/%s: %s", config.Model, rName.Subtype, config.Name, err)
```
Comment (Kschappacher) [inline robot/impl/local_robot.go:557]: Do we really want to remove all the identifying descriptors here? I can imagine a case where these errors may be hard to debug especially if you have two components with same subtype and model.

Comment (cheukt) [reply robot/impl/local_robot.go:557]: moved it to https://github.com/viamrobotics/rdk/pull/1514/files#diff-77fd7ff51c1b7804c67c4df0a9a672ea0e3842b4a54bbd8ef81af82ac95c4f22R368-R369 so all errors from that function gets the additional info

**Review by Kschappacher — APPROVED** [review]
One standing question but otherwise LGTM

File: robot/impl/resource_manager.go [hunk 2]
```diff
@@ -514,7 +520,11 @@ func (manager *resourceManager) RemoteByName(name string) (robot.Robot, bool) {
 	if iface, ok := manager.resources.Node(rName); ok {
 		part, ok := iface.(robot.Robot)
 		if !ok {
-			manager.logger.Errorw("tried to access remote but its not a robot interface", "remote_name", name, "type", iface)
+			if ph, ok := iface.(*resourcePlaceholder); ok {
+				manager.logger.Errorw("failed to get remote", "remote", name, "err", ph.err)
```
Comment (npmenard) [inline robot/impl/resource_manager.go:524]: nit: remote not available

File: robot/impl/resource_manager.go [hunk 3]
```diff
@@ -514,7 +520,11 @@ func (manager *resourceManager) RemoteByName(name string) (robot.Robot, bool) {
...
+			} else {
+				manager.logger.Errorw("tried to access remote but its not a robot interface", "remote", name, "type", iface)
```
Comment (npmenard) [inline robot/impl/resource_manager.go:526]: are we going to get the relevant Type info without using %T?

File: robot/impl/resource_manager.go [hunk 4]
```diff
@@ -630,6 +641,7 @@ func (manager *resourceManager) wrapResource(name resource.Name, config interfac
 		wrapper = &resourcePlaceholder{
 			real:   part,
 			config: config,
+			err:    errors.New("resource being initialized"),
```
Comment (npmenard) [inline robot/impl/resource_manager.go:644]: nit: resource not initialized yet

**Review by npmenard — APPROVED** [review]
LGTM couples of nits.

File: robot/impl/resource_manager.go (hunk 2 above)
Comment (cheukt) [reply robot/impl/resource_manager.go]: updated

File: robot/impl/resource_manager.go (hunk 4 above)
Comment (cheukt) [reply robot/impl/resource_manager.go]: updated

File: robot/impl/resource_manager.go [hunk 5]
```diff
@@ -564,6 +574,7 @@ func (manager *resourceManager) markChildrenForUpdate(ctx context.Context, rName
 		wrapper := &resourcePlaceholder{
 			real:   nil,
 			config: originalConfig,
+			err:    errors.New("resource marked for update"),
```
[the remark this answers is not in the document]
Comment (cheukt) [reply robot/impl/resource_manager.go]: updated

**Pull request merged**